### ISSUE TITLE
File creation does not respect umask

### ISSUE THREAD

**Issue by madduck** [pimutils/vdirsyncer#804]
I'm using `vdirsyncer` 0.16.7-2 (Debian), and discovered that files created during sync do not respect `umask`. I.e., despite `umask 022`, the `singlefile` created in my scenario will have a mode of 0600.

**Comment by vpzomtrrfrt** [comment]
From what I can tell, this seems like expected behavior. umask 022 only restricts the file to not have write access for group and other, which 600 complies with.

**Comment by madduck** [comment]
@vpzomtrrfrt you are right that 600 is not incompatible with umask 022. However, as you can verify with standard tools (shell, touch, ...), the umask usually gets subtracted from 777 (for directories) or 666 (for less), and so files should have mode 644 with umask 022.
